gdx/src/com/badlogic/gdx/graphics/glutils/ETC1.java

```diff
@@ -114,10 +129,13 @@ public void write (FileHandle file) {
 					write.write(buffer, 0, bytesToWrite);
 					writtenBytes += bytesToWrite;
 				}
+
+				write.close();
+
+				gzip.close();
+
 			} catch (Exception e) {
-				throw new GdxRuntimeException("Couldn't write PKM file to '" + file + "'", e);
-			} finally {
-				StreamUtils.closeQuietly(write);

```

Comment:
If anything between 123 (`gzip = new GZIPOutputStream(outputStream);`) and 133 `gzip.close();` throws an exception, the streams will be left unclosed. And even worse, in `write(FileHandle)`, if anything on 106 in `write(OutputStream)` throws an exception, the file stream won't get closed.